src/server/tiered_storage.cc

```diff
@@ -120,10 +139,23 @@ struct TieredStorage::PerDb {
     // Entries that were scheduled to write but have not completed yet.
     InflightMap enqueued_entries;
   };
+  // Big bin entries that were scheduled to write but have not completed yet.
+  absl::flat_hash_map<string_view, SingleRequest*> bigbin_enqueued_entries;
 
   BinRecord bin_map[kSmallBinLen];
 };
 
+void TieredStorage::PerDb::CancelAll() {
+  for (size_t i = 0; i < kSmallBinLen; ++i) {
+    bin_map[i].pending_entries.clear();
+    bin_map[i].enqueued_entries.clear();
```

Comment:
add a comment that it's safe to clear enqueued_entries because the completion callback will take care of value deletions